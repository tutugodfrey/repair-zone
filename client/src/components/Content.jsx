import React, { Component } from 'react';
import { connect } from 'react-redux';
import Div from './elementComponents/Div.jsx';

class Content extends Component {
  render() {
    const { TabContent } = this.props;
    console.log
    return <Div divId="content-area" divClass="" content={<TabContent />} />
  }
}

export const mapStateToProps = (state) => {

  const { TabContent } = state;
  return {
    TabContent,
  }
}
export default connect(mapStateToProps)(Content);