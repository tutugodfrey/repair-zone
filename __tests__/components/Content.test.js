import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import Content from '../../client/src/components/Content.jsx';

const mockStore = configureStore();
const initialState = {
  pageReducer: {},
}
const store = mockStore(initialState);

describe('<Content />', () => {
  const wrapper = shallow(
    <Content
      store={store}
    />
  );
  test('should match the snapshot of the component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
