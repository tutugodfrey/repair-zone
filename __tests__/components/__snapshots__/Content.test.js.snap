// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> should match the snapshot of the component 1`] = `
<Content
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
